Firestore/core/test/unit/model/value_util_test.cc

```diff
@@ -112,6 +114,25 @@ class ValueUtilTest : public ::testing::Test {
     }
   }
 
+  void VerifyRelaxedAscending(Message<google_firestore_v1_ArrayValue>& left,
+                              Message<google_firestore_v1_ArrayValue>& right) {
+    for (pb_size_t i = 0; i < left->values_count; ++i) {
+      for (pb_size_t j = 0; j < right->values_count; ++j) {
+        EXPECT_NE(ComparisonResult::Descending,
```

Comment:
FWIW, I think this is pretty clever but it took me a while to understand how this was meant to work.

Reply:
Done.